CompilerSource/compiler/components/write_object_access.cpp

```diff
@@ -104,26 +104,23 @@ int lang_CPP::compile_writeObjAccess(map<int,parsed_object*> &parsed_objects, pa
     wto << "  " << i->second.original.type << " " << i->second.original.prefix << "dummy_" << i->second.uc << i->second.original.suffix << "; // Referenced by " << uc << " accessors" << endl;
   }
 
-  for (map<string,dectrip>::iterator dait = dot_accessed_locals.begin(); dait != dot_accessed_locals.end(); dait++)
-  {
+  for (auto dait = dot_accessed_locals.begin(); dait != dot_accessed_locals.end(); dait++) {
     const string& pmember = dait->first;
     wto << "  " << dait->second.type << " " << dait->second.prefix << REFERENCE_POSTFIX(dait->second.suffix) << " &varaccess_" << pmember << "(int x)" << endl;
     wto << "  {" << endl;
 
     wto << "    object_basic *inst = fetch_instance_by_int(x);" << endl;
     wto << "    if (inst) switch (inst->object_index)" << endl << "    {" << endl;
 
-    for (po_i it = parsed_objects.begin(); it != parsed_objects.end(); it++)
-    {
-      parsed_object *parent = it->second;
+    for (parsed_object *parent : parsed_objects) {
       while (parent) {
         map<string,dectrip>::iterator x = parent->locals.find(pmember);
         if (x != parent->locals.end())
         {
           string tot = x->second.type != "" ? x->second.type : "var";
           if (tot == dait->second.type and x->second.prefix == dait->second.prefix and x->second.suffix == dait->second.suffix)
           {
-            wto << "      case " << it->second->name << ": return ((OBJ_" << it->second->name << "*)inst)->" << pmember << ";" << endl;
+            wto << "      case " << parent->name << ": return ((OBJ_" << parent->name << "*)inst)->" << pmember << ";" << endl;
```

Comment:
this is cause of bug @RobertBColton crying about. It appears when an object is used as a parent more than once. I'm not sure what the fix is

Reply:
Right you are. That was a hasty change on my part. Fixed.